sql: report the table writer in EXPLAIN

knz opened the pull request:
This patch changes EXPLAIN to report the internal algorithm used to
perform a mutation statement (INSERT etc). This can be used for
performance troubleshooting.

We are not documenting this since this is likely to change.

Release note: None

knz committed:
sql: report the table writer in EXPLAIN

This patch changes EXPLAIN to report the internal algorithm used to
perform a mutation statement (INSERT etc). This can be used for
performance troubleshooting.

We are not documenting this since this is likely to change.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/33389)
<!-- Reviewable:end -->

jordanlewis reviewed (commented):
What about making this only appear in `EXPLAIN(verbose)`? Code <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33389#-:-LUpySgL1M05frc9mMs7:b19tref)* status: :shipit: complete! 1 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

jordanlewis commented:
Could you add a `verboseAttr` field to the observer that checks the `explainer` state the way that `expr` does?

knz commented:
> Could you add a verboseAttr field to the observer that checks the explainer state the way that expr does?

I probably could -- but should I, just for that field?

jordanlewis commented:
Fair enough, I am just concerned that something like `deleter fast` will now be seen by clients and cause confusion.

Maybe changing the words would help. Here's my suggestion:

`s/writer/strategy`
`s/deleter fast/fast deleter`

knz reviewed (commented):
> Maybe changing the words would help. Here's my suggestion:
> s/writer/strategy
> s/deleter fast/fast deleter

Done. RFAL

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33389#-:-LVDGOw36jeOooA8Fyur:b-4bzylf)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

jordanlewis reviewed (approved):
I notice there's no `strict upserter` in the tests. Is that expected?

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33389#-:-LVEvYoG2c23FuATG7xJ:b-3ldc6z)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

knz committed:
sql: test the different insert/upsert strategies in logical plans

This commit adds the missing test that reveal the use of different
execution strategies depending on which INSERT/UPSERT statement is
used.

Release note: None

knz reviewed (commented):
> I notice there's no strict upserter in the tests. Is that expected?

Good call. Added.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/33389#-:-LVIT9Z982FYpU9n1LJE:bni71dk)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

knz commented:
bors r+
